### PR TITLE
Fix dynamo laravel cache store conditional check failed checks

### PR DESCRIPTION
**Pull request by GrahamCampbell**

### CODE REVIEW

File: src/Integration/Laravel/Cache/src/AsyncAwsDynamoDbStore.php [hunk 1]
```diff
@@ -233,13 +234,8 @@ public function add($key, $value, $seconds)
...
 
             return true;
-        } catch (HttpException $e) {
-            $type = $e->getAwsType();
-            if (null !== $type && Str::contains($type, 'ConditionalCheckFailedException')) {
-                return false;
-            }
-
-            throw $e;
+        } catch (ConditionalCheckFailedException $e) {
```
Comment (jderusse) [inline src/Integration/Laravel/Cache/src/AsyncAwsDynamoDbStore.php:237]: You need to bump `"async-aws/dynamo-db": "^1.1"` in composer.json to use this exception
(or let previous code to keep backward compatibility)

**Review by jderusse — COMMENTED** [review]
Thank you very much for working on it.
Just a small comment about BC

**Comment by GrahamCampbell** [comment]
Did the old code ever work? It isn't working for me with the latest package, hence this PR. :)

File: src/Integration/Laravel/Cache/src/AsyncAwsDynamoDbStore.php (hunk 1 above)
Comment (GrahamCampbell) [reply src/Integration/Laravel/Cache/src/AsyncAwsDynamoDbStore.php:236]: Done.

**Comment by Nyholm** [comment]
Yes, it did work when it was first created. I guess we are missing some tests for this. That is why we accidentally broke it when updating the exceptions.

**Review by jderusse — APPROVED** [review]
thank you.

**Review by Nyholm — APPROVED** [review]
Thank you

**Pull request merged**